packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -226,9 +226,17 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 
 		this.runtime.on("disconnected", () => this.onDisconnected());
 		this.redirectTable = this.load(snapshot);
-
 		// Begin uploading stashed blobs from previous container instance
+		console.log("constructor");
 		Object.entries(stashedBlobs).forEach(([localId, entry]) => {
+			if (entry.minTTLInSeconds && entry.serverUploadTime) {
+				console.log("DEBUG entry has TTL and was previously uploaded");
+				const timeLapseSinceServerUpload = (Date.now() - entry.serverUploadTime) / 1000;
+				if (entry.minTTLInSeconds - timeLapseSinceServerUpload > 0) {
```

Comment:
I think we should treat blobs that are about to expire the same as blobs that have already expired. I'm not sure what the best approach is, but something like uploading when more than half of the time has elapsed would give us most of the performance benefits. I think the main thing we want to avoid is reuploading when clients are switching frequently between documents, when maybe only a few minutes have passed.

Reply:
half-life is a good strategy, it keeps safety, and avoids doing some unnecessary work.